wordpress-mobile/WordPress-iOS
# Expose API to fetch All Domains

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/21556

## Description

This PR adds an API to execute All Domains request.

## Related PR
This PR depends on:
- https://github.com/wordpress-mobile/WordPressKit-iOS/pull/626

## Test Instructions

1. Paste this code snippet in `MySiteViewController.viewDidAppear`.
```swift
if let account = defaultAccount() {
    var params = DomainsService.AllDomainsEndpointParams()
    params.resolveStatus = true
    params.locale = "en"
    let service = DomainsService(coreDataStack: ContextManager.shared, account: account)
    service.fetchAllDomains(params: params) { result in
        switch result {
        case .success(let domains):
            print(domains)
        case .failure(let error):
            print(error)
        }
    }
}
```
2. Build and run the Jetpack app.
3. Expect `domains` to be printed in the console.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
